### ISSUE TITLE
[5.3.1] Supports Not Snug In Certain Cases

### ISSUE THREAD

**Issue by mzhu1113** [Ultimaker/Cura#15660]
### Application Version

5.3.1

### Operating System

W11

### Printer

N/A

### Reproduction steps

1. Use the example.stl with supports on, touching buildplate, normal.
2. Slice. Doesn't matter what settings you change, behavior doesn't go away.

### Actual results

Attached example file that generates extraneous supports. In more complex models, this issue would show up as supports that would not be "snug" and the generated support would be basically useless and waste of material.

Other than turning off supports, this behavior isn't controllable from the UI from what I can tell.

Because this issue was not a problem in prior version of Cura until 5.3, this is considered a regression.

![image](https://github.com/Ultimaker/Cura/assets/11019907/3f3fe781-5d8c-45df-8ebf-12e3f1de5234)
Left is 5.2.1, Right is 5.3.1
As portrayed in the picture, this behavior doesn't happen 100% of the time.

### Expected results

Snug supports, no extraneous supports generated from below where it isn't needed.

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### Additional information & file uploads
I couldn't find if this issue was reported already, apologies if it has.

Attachment:
[example.zip](https://github.com/Ultimaker/Cura/files/11636518/example.zip)

**Comment by GregValiant** [comment]
Thanks for the report.
I have your model but you didn't include a Project File so I don't have your settings.
These slices are with 5.3.1
![image](https://github.com/Ultimaker/Cura/assets/64202104/ef550af2-7b77-48b0-a22e-49f00b2fff1b)

![image](https://github.com/Ultimaker/Cura/assets/64202104/a9099170-f7cd-4c67-9923-2948f8153309)

The difference is that with "Towers" enabled and with a bit of "Support Horizontal Expansion" the tower shows up but doesn't connect to anything because the supports are configured for "Touching Build".

I was aware of this as I've seen it in my own slices when I forget to reset the profile after making changes specific to a model.  I'll leave this as a bug report as it certainly isn't clear that this can happen.
I had to make some guesses here.  If you would provide a Project File it would be really helpful.

**Comment by mzhu1113** [comment]
> Thanks for the report. I have your model but you didn't include a Project File so I don't have your settings. These slices are with 5.3.1 ![image](https://user-images.githubusercontent.com/64202104/243140212-ef550af2-7b77-48b0-a22e-49f00b2fff1b.png)
> 
> ![image](https://user-images.githubusercontent.com/64202104/243140257-a9099170-f7cd-4c67-9923-2948f8153309.png)
> 
> The difference is that with "Towers" enabled and with a bit of "Support Horizontal Expansion" the tower shows up but doesn't connect to anything because the supports are configured for "Touching Build".
> 
> I was aware of this as I've seen it in my own slices when I forget to reset the profile after making changes specific to a model. I'll leave this as a bug report as it certainly isn't clear that this can happen. I had to make some guesses here. If you would provide a Project File it would be really helpful.

Yes this seems to have been the issue. I've redone our profiles to disable these so this isn't a problem (for us) anymore but I believe this issue should be reclassified as an enhancement request to avoid creating these towers if there's no actual connection being made. Otherwise it is a a material waste.

Thanks for the help,

**Comment by GregValiant** [comment]
I've added the "Improvement" label.  I'm going to leave it as a bug report.  That way I know the Cura team will see it.
